pkg/controller/template_controller.go

```diff
@@ -0,0 +1,30 @@
+package controller
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/operator-framework/combo/api/v1alpha1"
+)
+
+type templateController struct {
+	client.Client
+	log logr.Logger
+}
+
+func (t *templateController) manageWith(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.Template{}).
+		Complete(t)
+}
+
+func (t *templateController) Reconcile(ctx context.Context, req ctrl.Request) (reconcile.Result, error) {
+	// Set up a convenient log object so we don't have to type request over and over again
+	log := t.log.WithValues("request", req)
+	log.V(1).Info("reconciling template")
```

Comment:
Any idea on how we'd be able to configure t.log's logging level using the manageable's implementation?

Reply:
This is what I came up with:

```go
// controller.go
...
func (m manageables) manageWith(mgr ctrl.Manager) error {
	for _, man := range m {
		if m == nil {
			// Panic because this case indicates a bug
			panic("failed to register controller with manager: cannot be nil")
		}

		if err := man.manageWith(mgr, 1); err != nil {
			// Bail out if any controller fails to register with the manager
			return err
		}
	}

	return nil
}
...
```

```go
// combo_controller.go
...
func (c *combinationController) manageWith(mgr ctrl.Manager, version int) error {
	c.log = c.log.V(version)
	return ctrl.NewControllerManagedBy(mgr).
		For(&v1alpha1.Combination{}).
		Complete(c)
}
...
```
What do you think?